packages/mermaid/src/mermaidAPI.ts

```diff
@@ -481,9 +462,7 @@ const render = async function (
   // Insert an element into svg. This is where we put the styles
   const svg = element.firstChild;
   const firstChild = svg.firstChild;
-  const diagramClassDefs = CLASSDEF_DIAGRAMS.includes(diagramType)
-    ? diag.renderer.getClasses(text, diag)
-    : {};
+  const diagramClassDefs = diag.renderer.getClasses?.(text, diag);
```

Comment:
This is the main logic change in the PR. Rest are mostly type changes.

Reply:
Would it be better to add a `?? {}`; to the this line, so it more closely matches the previous code.

That way, we won't need to add `| undefined` to the type.

```ts
  const diagramClassDefs = diag.renderer.getClasses?.(text, diag) ?? {};
```

Reply:
That was what I did first, but the actual place where the value is used already expects `undefined | null`. So there is no need to assign here. 

https://github.com/mermaid-js/mermaid/blob/e75af86ef25c3af7ee54ba04f39d38be5c7beea1/packages/mermaid/src/mermaidAPI.ts#L175-L178